exonum-java-binding-common/src/main/java/com/exonum/binding/common/proofs/list/ListProofElement.java

```diff
@@ -16,24 +16,40 @@
 
 package com.exonum.binding.common.proofs.list;
 
+import static com.google.common.base.Preconditions.checkNotNull;
+
 import com.exonum.binding.common.hash.Funnel;
 import com.exonum.binding.common.hash.PrimitiveSink;
+import com.google.protobuf.ByteString;
 
 /**
  * Represents an element of a proof list: a leaf node in a list proof tree.
  */
 public final class ListProofElement implements ListProof {
 
-  private final byte[] element;
+  private final ByteString element;
+
+
+  /**
+   * Creates a new ListProofElement.
+   *
+   * @param element an element of the list
+   * @throws NullPointerException if the element is null
+   */
+  // TODO: create ByteString in native
+  @SuppressWarnings("unused")  // Native API
+  ListProofElement(byte[] element) {
+    this.element = ByteString.copyFrom(element);
+  }
 
   /**
    * Creates a new ListProofElement.
    *
    * @param element an element of the list
    * @throws NullPointerException if the element is null
    */
-  public ListProofElement(byte[] element) {
-    this.element = element.clone();
+  ListProofElement(ByteString element) {
```

Comment:
It must be public as the type is public, otherwise, there is no way to instantiate it outside of the package.